### PR TITLE
Fixed issue "blockid shouldn't be converted to lower case in logs #1609"

### PR DESCRIPTION
**Pull request by ishaannverma**

### CODE REVIEW

**Comment by mohsha-msft** [comment]
Running CI.

File: common/extensions.go [hunk 1]
```diff
@@ -70,19 +70,30 @@ const SigAzure = azbfs.SigAzure
 const SigXAmzForAws = azbfs.SigXAmzForAws
 
 func RedactSecretQueryParam(rawQuery, queryKeyNeedRedact string) (bool, string) {
-	rawQuery = strings.ToLower(rawQuery) // lowercase the string so we can look for ?[queryKeyNeedRedact] and &[queryKeyNeedRedact]=
+	// make a dictionary of all the params map[param]value
+	copyQuery := rawQuery
+	originalDict := map[string]string{}
+	for _, param := range strings.Split(copyQuery, "&") {
```
Comment (mohsha-msft) [inline common/extensions.go:76]: Check this out: https://pkg.go.dev/net/url#example-ParseQuery
Do you think this will make code efficient and concise?

Comment (ishaannverma) [reply common/extensions.go:76]: Okay 👍

**Review by mohsha-msft — APPROVED** [review]
Changes look good to me. Can you also verify that AzCopy logs have SAS Signature redacted as well?

File: common/extensions_test.go [hunk 2]
```diff
@@ -84,3 +84,60 @@ func (*extensionsTestSuite) TestURLWithPlusDecodedInPath(c *chk.C) {
 		c.Assert(extension.String(), chk.Equals, v.expectedResult)
 	}
 }
+
+func (*extensionsTestSuite) TestRedaction(c *chk.C) {
+
+	// must make sure that-
+	//1. the signature is redacted if present
+	//2. the capitalization of the rest of the string should not be affected
+	//3. the order of the rest of the string should not be affected
```
Comment (mohsha-msft) [inline common/extensions_test.go:93]: order doesn't matter

**Review by nakulkar-msft — APPROVED** [review]

**Review by siminsavani-msft — APPROVED** [review]
Good job on the tests!

**Review by adreed-msft — APPROVED** [review]

**Pull request merged**